logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/SearchUserRepository.kt

```diff
@@ -132,23 +132,29 @@ internal class SearchUserRepositoryImpl(
             maxResultSize,
             searchUsersOptions
         ).flatMap {
-            wrapApiRequest {
-                userDetailsApi.getMultipleUsers(ListUserRequest.qualifiedIds(it.documents.map { it.qualifiedID }))
-            }.map { userDetailsResponses ->
-                val selfUser = getSelfUser()
-
-                UserSearchResult(userDetailsResponses.map { userProfileDTO ->
-                    publicUserMapper.fromUserDetailResponseWithUsertype(
-                        userDetailResponse = userProfileDTO,
-                        userType = userTypeMapper.fromTeamAndDomain(
-                            otherUserDomain = userProfileDTO.id.domain,
-                            selfUserTeamId = selfUser.teamId?.value,
-                            otherUserTeamId = userProfileDTO.teamId,
-                            selfUserDomain = selfUser.id.domain,
-                            isService = userProfileDTO.service != null,
+            val qualifiedIdList = it.documents.map { it.qualifiedID }
```

Comment:
will it be easier to do such logic on the network side

```kotlin
class MyApi {
    suspend fun fetchUserInfo(userIdList: List<UserId>) {
        if(userIdList.isEmpty) {
            return NetworkResponse.Sucess(emptyList())
        }
    }
}
```

Reply:
I didn't want to do it on network layer because this way we're faking the request and passing empty headers in response for instance which wouldn't be true in the real response returned by the backend, also maybe sometimes we would want to make such request and handle the potential error 🤔